app/controllers/course/survey/responses_controller.rb

```diff
@@ -37,11 +37,22 @@ def edit
   end
 
   def update
-    @response.submit if params[:response][:submit]
-    if @response.update_attributes(response_update_params)
-      render_response_json
-    else
-      render json: { errors: @response.errors }, status: :bad_request
+    if params[:response][:unsubmit]
+      @response.unsubmit if can?(:update, @survey)
+
+      if @response.update_attributes(response_unsubmit_params)
```

Comment:
> from @weiqingtoh Small concern: since unsubmit is more of a single action - however, it is actually possible for a user to pass in updated response params, which seems as though it would get updated here as well. Even if the react components don't pass these params into the controller, it might still be cleaner to ensure the controller doesn't do this.

`response_unsubmit_params` will not allow any of the form params to update